librz/core/cil.c

```diff
@@ -475,6 +475,10 @@ RZ_IPI bool rz_core_analysis_il_vm_set(RzCore *core, const char *var_name, ut64
 	case RZ_IL_TYPE_PURE_BOOL:
 		val = rz_il_value_new_bool(rz_il_bool_new(value != 0));
 		break;
+	case RZ_IL_TYPE_PURE_FLOAT:
+		// TODO : ut64 value is enough for user input ?
```

Comment:
We could use union instead. @thestr4ng3r what do you think?

Reply:
I don't think ut64 makes sense for float at all, probably the parsing of the string should be pulled in here and made dependent on the variable sort.

Reply:
i think instead the value should be set by a new function or we modify the function to receive a pointer and based on the type we cast it in double or ut64

Reply:
for example:
```c
RZ_IPI bool rz_core_analysis_il_vm_set(RzCore *core, const char *var_name, void *pvalue) {
	switch (var->sort.type) {
	case RZ_IL_TYPE_PURE_BITVECTOR:
		val = rz_il_value_new_bitv(rz_bv_new_from_ut64(var->sort.props.bv.length, *((ut64*)pvalue)));
		break;
	case RZ_IL_TYPE_PURE_BOOL:
		val = rz_il_value_new_bool(rz_il_bool_new(*((ut64*)pvalue) != 0));
		break;
	case RZ_IL_TYPE_PURE_FLOAT:
		// TODO: handle the right float type with a switch
		val = rz_il_value_new_bool(rz_il_float_new(RZ_FLOAT_IEEE754_BIN_64, *((double*)pvalue)));
		break;
```

Reply:
makes sense